src/main/java/io/confluent/kafkarest/v2/KafkaConsumerTopicState.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.kafkarest.v2;
+
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import io.confluent.kafkarest.v2.KafkaConsumerReadTask;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+
+/**
+ * Tracks a consumer's state for a single topic, including the underlying stream and consumed and
+ * committed offsets. It provides manual synchronization primitives to support KafkaConsumerWorkers
+ * protecting access to the state while they process a read request in their processing loop.
+ */
+public class KafkaConsumerTopicState<KafkaK, KafkaV, ClientK, ClientV> {
```

Comment:
Given the suggested changes to the APIs and avoiding the topic URLs under a consumer in favor of just making subscription an explicit step (or part of instance creation), I think we can probably drop this class since each consumer will only need one instance of each of the pieces of data (i.e. the variables can probably just be lifted into `KafkaConsumerState`).